pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -215,40 +215,50 @@ func (os *OpenStack) AttachDisk(instanceID, volumeID string) (string, error) {
 	if err != nil {
 		return "", err
 	}
+
 	if volume.Status != VolumeAvailableStatus {
```

Comment:
I don't understand your point. If the status is Available, it will continue with the attach loop, otherwise it won't... IIRC the attach/detach controller will then retry